spring-cloud-starter-stream-source-mongodb/src/test/java/org/springframework/cloud/stream/app/mongodb/source/MongodbSourceApplicationTests.java

```diff
@@ -16,121 +16,176 @@
-			assertThat((String)received.getPayload(), CoreMatchers.containsString("hola"));
+			Message<?> received =
+					this.messageCollector
+							.forChannel(this.source.output())
+							.poll(2, TimeUnit.SECONDS);
+			assertThat(received, notNullValue());
+			assertThat((String) received.getPayload(), containsString("hola"));
 		}
+
 	}
 
-	@IntegrationTest(value = {"mongodb.collection=testing", "mongodb.query={ 'greeting': 'bogus' }", "trigger.fixedDelay=1"})
+	@TestPropertySource(properties = {
+			"mongodb.query={ 'greeting': 'bogus' }",
+			"trigger.fixedDelay=1" })
 	public static class InvalidQueryTests extends MongodbSourceApplicationTests {
+
 		@Test
 		public void test() throws InterruptedException {
-			Message<?> received = messageCollector.forChannel(source.output()).poll(2, TimeUnit.SECONDS);
-			assertThat(received, CoreMatchers.nullValue());
+			Message<?> received =
+					this.messageCollector
+							.forChannel(this.source.output())
+							.poll(2, TimeUnit.SECONDS);
+			assertThat(received, nullValue());
 		}
+
 	}
 
-	@IntegrationTest(value = {"mongodb.collection=testing", "trigger.fixedDelay=1", "mongodb.split=false"})
+	@TestPropertySource(properties = {
+			"trigger.fixedDelay=1",
+			"mongodb.split=false" })
 	public static class NoSplitTests extends MongodbSourceApplicationTests {
+
 		@Test
 		public void test() throws InterruptedException {
-			Message<?> received = messageCollector.forChannel(source.output()).poll(2, TimeUnit.SECONDS);
-			assertThat(received, CoreMatchers.notNullValue());
-			assertThat(received.getPayload(), Matchers.instanceOf(List.class));
-			assertThat(received.getPayload().toString(), CoreMatchers.containsString("hola"));
-			assertThat(received.getPayload().toString(), CoreMatchers.containsString("hello"));
+			Message<?> received =
+					this.messageCollector
+							.forChannel(this.source.output())
+							.poll(2, TimeUnit.SECONDS);
+			assertThat(received, notNullValue());
+			assertThat(received.getPayload(), instanceOf(List.class));
+			assertThat(received.getPayload().toString(), containsString("hola"));
+			assertThat(received.getPayload().toString(), containsString("hello"));
 		}
+
 	}
 
-	@IntegrationTest(value = {"mongodb.collection=testing", "trigger.fixedDelay=100"})
+	@TestPropertySource(properties = "trigger.fixedDelay=100")
 	public static class MongoTriggerTests extends MongodbSourceApplicationTests {
+
 		@Test
 		public void test() throws InterruptedException {
-			Message<?> received = messageCollector.forChannel(source.output()).poll(1, TimeUnit.SECONDS);
-			assertThat(received, CoreMatchers.nullValue());
+			Message<?> received =
+					this.messageCollector
+							.forChannel(this.source.output())
+							.poll(1, TimeUnit.SECONDS);
+			assertThat(received, nullValue());
 		}
+
 	}
 
+	@TestPropertySource(properties = {
+			"mongodb.query-expression=new BasicQuery('{ }')" +
+					".limit(1)" +
+					".with(new org.springframework.data.domain.Sort('greeting'))",
```

Comment:
@garyrussell , this `Sort` sample is for discussion.
I would like to include this `org.springframework.data.domain` package to SI by default here in the `MongoDbMessageSource`:
```
//Register MongoDB query API package so FQCN can be avoided in query-expression.
((StandardTypeLocator) typeLocator).registerImport("org.springframework.data.mongodb.core.query");
```
WDYT?